src/main/java/Duke.java

```diff
@@ -21,41 +12,99 @@ public static void main(String[] args) {
         while (isRunning) {
             userInput = in.nextLine();
             String[] userInputSplit = userInput.split(" ");
-            Task task;
-
-            switch (userInputSplit[0]) {
-            case "bye":
-                System.out.println("Bye. Hope to see you again soon!");
-                isRunning = false;
-                break;
-            case "list":
-                System.out.println("Here are the tasks in your list:");
-                for (int i = 0; i < listOfTasks.size(); i++) {
-                    task = listOfTasks.get(i);
-                    System.out.println(i + 1 + "." + task.description);
-                }
-                break;
-            case "mark":
-                int markTask = Integer.parseInt(userInputSplit[1]) - 1;
-                task = listOfTasks.get(markTask);
-                task.markAsDone();
-                System.out.println("Nice! I've marked this task as done:");
-                System.out.println(task.getDescription());
-                break;
-            case "unmark":
-                int unmarkTask = Integer.parseInt(userInputSplit[1]) - 1;
-                task = listOfTasks.get(unmarkTask);
-                task.markAsUndone();
-                System.out.println("OK, I've marked this task as not done yet: ");
-                System.out.println(task.getDescription());
-                break;
-            default:
-                Task newTask = new Task("[ ] " + userInput);
-                listOfTasks.add(newTask);
-                System.out.println("added: " + userInput);
-                break;
+            isRunning = runCommand(userInput, listOfTasks, userInputSplit);
+        }
+    }
```

Comment:
Indentation issues. Since line +15 is within the switch statement, the closing bracket of switch should be aligned with it. Similar issue with closing bracket of while loop on line +17